File: starter_kit/bots/python/example_bots/kraken_bot.py
def do_turn(game):
    kraken = game.get_kraken()
    if not kraken:
        game.debug("No Kraken!")
        return

    if (game.get_turn() == 1):
        game.debug("Kraken settings: Turns per move: %s, awake turns: %s, sleep turns: %s, vanished turns: %s" %
            (game.get_kraken_turns_per_move(),
            game.get_kraken_awake_turns(),
            game.get_kraken_sleep_turns(),
            game.get_kraken_vanished_turns()))

    if kraken:
        game.debug("Kraken: %s can_move: %s" % (kraken, kraken.can_move))

    num_pirates = len(game.my_pirates())
    total_pirates = len(game.all_my_pirates())

    if num_pirates == 0:
        return

    for pirate in game.my_pirates():
        if (pirate.id == 0):
            if kraken.is_asleep:
                if kraken.location == pirate.location:
                    game.unleash_the_kraken(pirate, (10,10))
                else:
                    directions = game.get_directions(pirate, kraken)
                    game.set_sail(pirate, directions[0])

            continue

        row = game.get_rows() / total_pirates * pirate.id
        if (pirate.id % 2 == 0):
            row = game.get_rows() - 1 - game.get_rows() / total_pirates * pirate.id
        col = game.get_cols() / total_pirates * pirate.id
        dest = (row, col)
        game.debug("sending pirate %s to %s" %(pirate.id, dest))
        directions = game.get_directions(pirate, dest)
        next_location = game.destination(pirate, directions[0])
        if not game.safe_from_kraken(next_location):
            game.debug("Sending pirate %s to possible death..." % [pirate.id])

        game.set_sail(pirate, directions[0])

File: starter_kit/bots/csharp/example_bots/KrakenBot.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Pirates;


namespace HunterBot
{
    public class KrakenBot : Pirates.IPirateBot
	{

        public void DoTurn(IPirateGame game)
        {
            Kraken kraken = game.GetKraken();
		
		    if (kraken == null) {
			    game.Debug("No Kraken!");
	            return;
		    }
		
		if (game.GetTurn() == 1) {
            game.Debug("Playing vs {0}", game.GetOpponentName());
            game.Debug("Kraken settings: Turns per move: {0}, awake turns: {1}, sleep turns: {2}, vanished turns: {3}",
	                game.GetKrakenTurnsPerMove(),
	                game.GetKrakenAwakeTurns(),
	                game.GetKrakenSleepTurns(),
	                game.GetKrakenVanishedTurns());
		}
        game.Debug("Kraken: {0}", kraken);
        
		if (kraken.State == KrakenState.VANISHED) {
			game.Debug("WHERE DID THE KRAKEN GO???");
		}
		
		int numPirates = game.MyPirates().Count;
		int totalPirates = game.AllMyPirates().Count;
		if (numPirates == 0) {
			return;
		}

        foreach (Pirate pirate in game.MyPirates())
        {
            if (pirate.Id == 0)
            {
                if (kraken.IsAsleep)
                {
                    if (kraken.Loc.Equals(pirate.Loc))
                    {
                        game.UnleashTheKraken(pirate, new Location(10, 10));
                    }
                    else
                    {
                        List<Direction> dirs = game.GetDirections(pirate, kraken.Loc);
                        game.SetSail(pirate, dirs[0]);
                    }
                }

                continue;
            }

            // Just arrange the other pirates to see that the kraken approach them
            int row = game.GetRows() / totalPirates * pirate.Id;
            if (pirate.Id % 2 == 0)
            {
                // Put some of the pirates inverse locations so they wouldn't be on the same line;
                row = game.GetRows() - 1 - game.GetRows() / totalPirates * pirate.Id;
            }
            int col = game.GetCols() / totalPirates * pirate.Id;

            Location dest = new Location(row, col);
            game.Debug("sending pirate {0} to {1}", pirate.Id, dest);
            List<Direction> directions = game.GetDirections(pirate, dest);
            Location destinationLoc = game.Destination(pirate, directions[0]);
            if (!game.SafeFromKraken(destinationLoc))
            {
                game.Debug("Sending pirate {0} to possible death...", pirate.Id);
            }
            game.SetSail(pirate, directions[0]);
        }
        }
	}
}

